File: Thesis/code/discord.java
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
import net.dv8tion.jda.api.hooks.ListenerAdapter;
import net.dv8tion.jda.api.interactions.components.buttons.Button;

public void onSlashCommandInteraction(SlashCommandInteractionEvent event) {
    if (event.getName().equals("pow")) {
        int num = event.getOption("num").getAsInt();
        int exp = event.getOption("exp").getAsInt();
        event.reply("" + Math.pow(num,exp)).addActionRow(Button.link("https://example.com", "Example")).queue();
    }
}

File: Thesis/code/discord.cs
using Discord;
using Discord.WebSocket;

private async Task SlashCommandHandler(SocketSlashCommand command) {
    if (command.Data.Name == "pow") {
        var num = (Int64)command.Data.Options.ElementAt(0).Value;
        var exp = (Int64)command.Data.Options.ElementAt(1).Value;
        var builder = new ComponentBuilder().WithButton("Example", style: ButtonStyle.Link, url: "https://example.com");
        await command.RespondAsync("" + Math.Pow(num, exp), components: builder.Build());
    }
}